packages/roosterjs-editor-dom/lib/table/VTable.ts

```diff
@@ -120,6 +145,237 @@ export default class VTable {
+                    this.formatBorders(
+                        format,
+                        cell.td,
+                        rowIndex === 0,
+                        rowIndex === this.cells.length - 1,
+                        cellIndex === 0,
+                        cellIndex === this.cells[0].length - 1,
+                        rowIndex === this.cells[0].length - 1
+                    );
+                }
+            });
+        });
+    }
+
+    /**
+     * Apply custom design to the first table column
+     * @param format
+     * @returns
+     */
+    private setFirstColumnFormat(format: Partial<TableFormat>) {
+        if (!format.hasFirstColumn) {
+            this.forEachCellOfColumn(0, (cell, row, i) => {
+                if (cell.td) {
+                    cell.td = changeElementTag(
+                        cell.td,
+                        TABLE_CELL_TAG_NAME
+                    ) as HTMLTableCellElement;
+                    cell.td.scope = '';
+                }
+            });
+            return;
+        }
+        this.forEachCellOfColumn(0, (cell, row, i) => {
+            if (cell.td) {
+                if (i !== 0) {
+                    cell.td.style.borderTopColor = TRANSPARENT;
+                    cell.td.style.backgroundColor = TRANSPARENT;
+                }
+                if (i !== this.cells.length - 1 && i !== 0) {
+                    cell.td.style.borderBottomColor = TRANSPARENT;
+                }
+                cell.td = changeElementTag(cell.td, TABLE_HEADER_TAG_NAME) as HTMLTableCellElement;
+                cell.td.scope = 'col';
+            }
+        });
+    }
+
+    /**
+     * Apply custom design to the Header Row
+     * @param format
+     * @returns
+     */
+    private setHeaderRowFormat(format: TableFormat) {
+        if (!format.hasHeaderRow) {
+            this.forEachCellOfRow(0, (cell, i) => {
+                if (cell.td) {
+                    cell.td = changeElementTag(
+                        cell.td,
+                        TABLE_CELL_TAG_NAME
+                    ) as HTMLTableCellElement;
+                    cell.td.scope = '';
+                }
+            });
+            return;
+        }
+        this.forEachCellOfRow(0, (cell, i) => {
+            if (cell.td) {
+                cell.td.style.backgroundColor = format.headerRowColor;
+                cell.td.style.borderRightColor = format.headerRowColor;
+                cell.td.style.borderLeftColor = format.headerRowColor;
+                cell.td.style.borderTopColor = format.headerRowColor;
+                cell.td = changeElementTag(cell.td, TABLE_HEADER_TAG_NAME) as HTMLTableCellElement;
+                cell.td.scope = 'row';
+            }
+        });
+    }
+
+    /**
+     * Save the format info of a table
+     * @param table The table the info will be saved
+     * @param format The format of the table
+     */
+    saveTableInfo(table: HTMLTableElement, format: TableFormat) {

```

Comment:
Move this function out. It can be put together with getTableFormatInfo. So you only need to define the TABLE_STYLE_INFO const once